scripts/maintenance/migrate_project/src/db.py

```diff
@@ -0,0 +1,163 @@
+    statemet = select([projects]).where(projects.c.uuid == f"{project_uuid}")
+    return connection.execute(statemet)
+
+
+def _get_hidden_project(connection: Connection, prj_owner: int) -> ResultProxy:
+    statemet = select([projects]).where(
+        and_(projects.c.prj_owner == prj_owner, projects.c.hidden == True)
+    )
+    return connection.execute(statemet)
+
+
+def _get_file_meta_data_without_soft_links(
+    connection: Connection, user_id: int, project_id: UUID
+) -> ResultProxy:
+    statemet = select([file_meta_data]).where(
+        and_(
+            file_meta_data.c.user_id == f"{user_id}",
+            file_meta_data.c.project_id == f"{project_id}",
+            file_meta_data.c.is_soft_link != True,
+        )
+    )
+    return connection.execute(statemet)
+
+
+def get_project_and_files_to_migrate(
+    project_uuid: UUID,
+    hidden_projects_for_user: Optional[int],
+    src_conn: Connection,
+    dst_conn: Connection,
+) -> Tuple[Deque, Deque]:
+    skipped_projects = deque()
+    skipped_files_meta_data = deque()
+
+    projects_to_migrate = deque()
+    files_meta_data_to_migrate = deque()
+
+    user_project_selection = list(_get_project(src_conn, project_uuid))
+    assert len(user_project_selection) == 1
+    user_project = user_project_selection[0]
+
+    project = dict(user_project.items())
+    project_id = project["uuid"]
+
+    if _project_uuid_exists_in_destination(dst_conn, project_id):
+        print(f"\n\n>>>>>>>>>SKIPPING, sync for {project=}\n\n")
+        skipped_projects.append(project)
+        raise Exception("could not continue, main project already exists on remote!")
+
+    projects_to_migrate.append(project)
+
+    if hidden_projects_for_user:
+        # extract all hidden projects and check if they require syncing
+        hidden_projects_cursor = _get_hidden_project(src_conn, hidden_projects_for_user)
+        for hidden_result in hidden_projects_cursor:
+            hidden_project = dict(hidden_result.items())
+            if _project_uuid_exists_in_destination(dst_conn, hidden_project["uuid"]):
+                print(f"\n\n>>>>>>>>>SKIPPING, sync for {project=}\n\n")
+                skipped_projects.append(project)
+                continue
+
+            projects_to_migrate.append(hidden_project)
+
+    # check file_meta_data in the projects to migrate
+    for project in projects_to_migrate:
+        user_id = project["prj_owner"]
+        project_id = project["uuid"]
+
+        files_metadata_cursor = _get_file_meta_data_without_soft_links(
+            connection=src_conn, user_id=user_id, project_id=project_id
+        )
+        for result in files_metadata_cursor:
+            file_meta_data = dict(result.items())
+            file_uuid = file_meta_data["file_uuid"]
+
+            if _meta_data_exists_in_destination(dst_conn, file_uuid):
+                print(f"\n\n>>>>>>>>>SKIPPING, sync for {file_meta_data=}\n\n")
```

Comment:
I would use either loggers or typer.echo (i think the latter can be also capture the former).... but no prints

Reply:
and typer can also use colors... which are nice when there are such things that are unexpected...

Reply:
Let's go for the typer echo and replace all prints.